### PR TITLE
pilot: fix to ensure we only use mesh config after it is initialized

### PR DESCRIPTION
**Pull request by howardjohn**
Fixes https://github.com/istio/istio/issues/52803

### CODE REVIEW

**Review by MorrisLaw — APPROVED** [review]
Lgtm. One question for my understanding.

It seems like `s.initMeshConfiguration` would mean initializing the mesh config. But is:
```go
s.initMeshNetworks(args, s.fileWatcher)
s.initMeshHandlers(configGen.MeshConfigChanged)
s.environment.Init()
if err := s.environment.InitNetworksManager(s.XDSServer); err != nil {
	return nil, err
}
```
considered part of the mesh config initialization which is why we add the trust bundle logic after this chunk instead of right after `s.initMeshConfiguration`?

**Comment by howardjohn** [comment]
> considered part of the mesh config initialization which is why we add the trust bundle logic after this chunk instead of right after `s.initMeshConfiguration`?

It doesn't really matter too much as long as its after initMeshConfiguration and before we need TrustBundle

File: pilot/pkg/bootstrap/server.go [hunk 1]
```diff
@@ -242,14 +242,12 @@ func NewServer(args *PilotArgs, initFuncs ...func(*Server)) (*Server, error) {
 		istiodCertBundleWatcher: keycertbundle.NewWatcher(),
 		webhookInfo:             &webhookInfo{},
 	}
-	s.workloadTrustBundle = tb.NewTrustBundle(nil, e.Watcher)
 
 	// Apply custom initialization functions.
 	for _, fn := range initFuncs {
 		fn(s)
 	}
 	// Initialize workload Trust Bundle before XDS Server
```
Comment (ramaraochavali) [inline pilot/pkg/bootstrap/server.go:250]: Move this comment also closer to line 289 and mention this needs to happen after mesh config is initialized

File: pilot/pkg/bootstrap/server.go [hunk 2]
```diff
@@ -242,14 +242,12 @@ func NewServer(args *PilotArgs, initFuncs ...func(*Server)) (*Server, error) {
 		istiodCertBundleWatcher: keycertbundle.NewWatcher(),
 		webhookInfo:             &webhookInfo{},
 	}
-	s.workloadTrustBundle = tb.NewTrustBundle(nil, e.Watcher)
```
Comment (hzxuzhonghu) [inline pilot/pkg/bootstrap/server.go:245]: I am confused, isn't moving back make initialization later than now?

Comment (hzxuzhonghu) [reply pilot/pkg/bootstrap/server.go:245]: got it, add some comments

**Review by howardjohn — COMMENTED** [review]
added comments

**Review by ramaraochavali — APPROVED** [review]

**Pull request merged**